integration-tests/slack/message-templates.js

```diff
@@ -0,0 +1,61 @@
+const {
+  CIRCLE_JOB,
+  CIRCLE_BUILD_NUM,
+  CIRCLE_SHA1
+} = process.env
+
+const getHeaderBlock = numFailedTests => {
+  if (numFailedTests === 0) {
+    return {
+      type: 'header',
+      text: {
+        type: 'plain_text',
+        text: ':circleci-pass:  Terra-UI Test has Passed on CircleCI'
+      }
+    }
+  }
+
+  return {
+    type: 'header',
+    text: {
+      type: 'plain_text',
+      text: ':circleci-fail:  Terra-UI Test has Failed on CircleCI'
+    }
+  }
+}
+
+const getJobDetailBlock = () => {
+  return {
+    type: 'section',
+    fields: [
+      {
+        type: 'mrkdwn',
+        text: `*Job ${CIRCLE_BUILD_NUM}*: <https://circleci.com/gh/DataBiosphere/terra-ui/${CIRCLE_BUILD_NUM} | ${CIRCLE_JOB}>`
+      },
+      {
+        type: 'mrkdwn',
+        text: `*Commit*: <https://github.com/DataBiosphere/terra-ui/commit/${CIRCLE_SHA1} | ${`${CIRCLE_SHA1}`.slice(0, 7)}>`
+      }
+    ]
+  }
+}
+
+const getTestsListBlock = testNames => {
+  return {
+    type: 'section',
+    fields: [
+      {
+        type: 'mrkdwn',
+        text: `*  ${testNames.join('\n*  ')}`
+      }
+    ]
+  }
+}
+
+// any non-zero number to indicate test(s) has failed in circleci job
+const getMessageBlockTemplate = (numFailedTests = 0) => [
+  getHeaderBlock(numFailedTests),
+  getJobDetailBlock()
+]
+
+module.exports = { getMessageBlockTemplate, getTestsListBlock }
```

Comment:
Do these need to be separated? Could `getMessageBlockTemplate` accept the list of failed tests instead of the number of failed tests and add the tests list block?

Reply:
No, doesn't need to be separated. For me, it's clean and clear. 
Below is what I came up with for `getMessageBlockTemplate(testNames)`. It works and one less func to export. But it's more code. Do you recommend to change to this?
```
const getMessageBlockTemplate = (testNames) => {
  const size = testNames.length

  const blocksArray = [
    getHeaderBlock(size),
    getJobDetailBlock()
  ]

  if (size === 0) {
    return blocksArray
  } else {
    return blocksArray.concat(getTestsListBlock(testNames))
  }
}
```

Reply:
I think I would be in favor of that change. That code has to go somewhere, and this way the logic for building the message is encapsulated in this module.

Reply:
That's a good point.  Updated and thank you.